assets/js/googlesitekit/modules/datastore/modules.js

```diff
@@ -1361,14 +1361,16 @@ const baseSelectors = {
 		}
 
 		return Object.keys( modules ).reduce( ( acc, slug ) => {
-			if ( modules[ slug ].shareable && ! modules[ slug ].internal ) {
-				if (
-					slug === 'analytics' &&
-					select( MODULES_ANALYTICS ).isGA4DashboardView()
-				) {
-					return { 'analytics-4': modules[ 'analytics-4' ], ...acc };
-				}
+			// When analytics is not connected, it is set to not shareable. So moving this if block
+			// within the shareable and internal check below will cause issues in that case.
```

Comment:
I think it's worth adding a bit more detail here to clarify what "issues" refers to. How does this look to you?

```suggestion
			// When Analytics is not connected, it is set to not shareable. So moving this if block
			// within the shareable and internal check below will cause issues in that case. Specifically,
			// when Analytics is active with GA4 connected, but UA is not connected, the Analytics 4 module
			// would not be included in the shareable modules list when it should be.
```